### PR TITLE
feat: subnetSelector-enhancement

### PR DESCRIPTION
**Pull request by smcavallo**
Signed-off-by: smcavallo <smcavallo@hotmail.com>

Fixes # 

**Description**
Support comma separated tag values in `subnetSelector` - (wildcard search is not flexible enough.)
Consider the use case where `Private1,Private2` subnets should be used but `Private3` excluded.
This is a simple enough change to split the string and supported by the ec2 tag filters.
It is done this same exact way for `aws-ids` searches.

**How was this change tested?**

* Unit tests

**Does this change impact docs?**
- [X ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ellistarn — DISMISSED** [review]
This is slick -- I love that this is innately supported by the AWS APIs.

File: pkg/cloudprovider/subnets_test.go [hunk 1]
```diff
@@ -17,8 +17,6 @@ package cloudprovider
 import (
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
-	. "github.com/onsi/ginkgo/v2"
-	. "github.com/onsi/gomega"
```
Comment (bwagner5) [inline pkg/cloudprovider/subnets_test.go:21]:
```suggestion
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"
```

File: pkg/cloudprovider/subnets_test.go [hunk 2]
```diff
@@ -27,7 +25,6 @@ import (
 
 	"github.com/aws/karpenter-core/pkg/apis/provisioning/v1alpha5"
 	coretest "github.com/aws/karpenter-core/pkg/test"
-	. "github.com/aws/karpenter-core/pkg/test/expectations"
```
Comment (bwagner5) [inline pkg/cloudprovider/subnets_test.go:30]:
```suggestion
	. "github.com/aws/karpenter-core/pkg/test/expectations"
```

**Review by ellistarn — APPROVED** [review]

**Comment by ellistarn** [comment]
Looks like this is failing tests: https://github.com/aws/karpenter/actions/runs/3489726900/jobs/5841741532

**Comment by smcavallo** [comment]
There is something wrong with the tests.  I am running this test with the original code and it is failing for the same reason.
The below should 100% have worked with the original code.
```
	It("should discover subnets by tag", func() {
		provider.SubnetSelector = map[string]string{"Name": "subnet-test2"}
		ExpectApplied(ctx, env.Client, test.Provisioner(coretest.ProvisionerOptions{Provider: provider}))
		pod := ExpectProvisioned(ctx, env.Client, recorder, controller, prov, coretest.UnschedulablePod())[0]
		ExpectScheduled(ctx, env.Client, pod)
		createFleetInput := fakeEC2API.CalledWithCreateFleetInput.Pop()
		Expect(fake.SubnetsFromFleetRequest(createFleetInput)).To(ConsistOf(
			"subnet-test2",
		))
	})
```
Looking into why....and will submit a fix....

**Comment by smcavallo** [comment]
@ellistarn - thank you!!!!  Should be all fixed now and the tests should all be passing.  My confusion was from how close the subnet `Name` and `id` are.  I tried to add a comment in the tests to help anyone else who is debugging.  It took a bit to figure out where the fixture info was coming from, but all set now.

**Review by ellistarn — APPROVED** [review]

**Comment by ellistarn** [comment]
One last thing @smcavallo, can you make an integration test in github.com/aws/karpenter/test/suites/integration/subnet_test.go. 

You can verify the test by running `FOCUS="testname from It() block" make e2etests`, which will run your singular test against a local cluster. Make sure you have your kubeconfig setup to point to somewhere where this test can execute correctly.

**Review by ellistarn — CHANGES_REQUESTED** [review]
Great work! Needs a test, though.

**Comment by smcavallo** [comment]
@ellistarn - thanks!!!  I took a shot at writing an integration test if you can please take a look.

File: test/suites/integration/subnet_test.go [hunk 3]
```diff
@@ -108,3 +131,35 @@ func getSubnets(tags map[string]string) map[string][]string {
 	Expect(err).To(BeNil())
 	return subnets
 }
+
+// SubnetInfo is a simple struct for testing
+type SubnetInfo struct {
```
Comment (ellistarn) [inline test/suites/integration/subnet_test.go:136]: Consider just passing the ec2 subnets around, rather than making another struct.

**Review by ellistarn — APPROVED** [review]
Nice!

**Comment by ellistarn** [comment]
@smcavallo, can you put the output of a successful test run in the PR description?

**Comment by tzneal** [comment]
Looks like it's just a go-lint issue renaming Id to ID is all that's remaining.

**Comment by smcavallo** [comment]
thanks @tzneal!!!! It has been updated.

**Review by ellistarn — APPROVED** [review]
Great work! thanks!

**Comment by smcavallo** [comment]
I just rebased off master to fix the tests which were referencing deprecated properties

**Review by ellistarn — DISMISSED** [review]

**Comment by ellistarn** [comment]
Hey @smcavallo , let me know if you want me to take over this PR to get it across the finish line.

**Comment by smcavallo** [comment]
@ellistarn - YES PLEASE, that would be great - thank you so much!!!!

**Comment by jonathan-innis** [comment]
Closed in favor of #3080

**Pull request closed without merging**